libtransmission/peer-mgr.cc

```diff
@@ -859,6 +865,7 @@ class tr_swarm
             (*it)->do_purge = true;
 
             // Note that it_that is invalid after this point
+            --stats.known_peer_from_count[info_that.from_first()];
```

Comment:
*(minor nit)* Please move this line above the comment so that the commment remain right above `connectable_pool.extract(it_that)`.